CONTRIBUTING.md

```diff
@@ -47,6 +47,93 @@ Using commit signoffs tags is mandatory for all commits; we also
 encourage that each commit is small and cohesive. See the next sections for details.
 
 
+### Programming style
+
+#### Code style
+
+Our code style follows [the Google C++ Style
+Guide](https://google.github.io/styleguide/cppguide.html) with a few
+exceptions. To run automatic code style formatting on files you modify, use our
+[clang-format
+template](https://github.com/mendersoftware/mendertesting/blob/master/.clang-format):
+
+```bash
+curl -O https://github.com/mendersoftware/mendertesting/raw/master/.clang-format
+clang-format -i --style=file:.clang-format <MODIFIED_FILES>
+```
+
+Note that the template requires clang-format version 15 or higher.
+
+##### C++ standard
+
+All shared code must use C++ features from no later than the C++11
+standard. Platform code for POSIX platforms is allowed to use C++17 if
+necessary, but C++11 is preferred for consistency with the rest of the code.
+
+#### File names
+
+Files must be named with `.cpp` and `.hpp` extensions. Only use `.h` if the
+header holds C declarations.
+
+##### Indentation
+
+Indentation in Mender code uses 1 tab per level, no spaces. Function arguments
+or initializers that are broken over multiple lines are indented one level.
+
+##### Line length
+
+Our line lengths are capped at 100 characters instead of 80, which is Google's
+cap. When considering line lengths, each tab is considered 4 characters
+wide. [The other guidelines for line
+length](https://google.github.io/styleguide/cppguide.html#Line_Length) still
+apply.
+
+##### Namespaces
+
+All the [rules about
+namespaces](https://google.github.io/styleguide/cppguide.html#Namespaces) apply,
+with one exception: It is allowed to use `using namespace std` to import the
+`std` namespace. All other namespaces must be spelled out where they are used
+(though it is allowed to shorten long names using `namespace short_name =
+long_name`).
```

Comment:
It reads to me like two exceptions instead of one: `namespace std` and `short_name = long_name`. I suggest we rewrite it to be more explicit.

For example:
All the rules about namespaces apply, with two exceptions. All namespaces must be spelled out where they are used except for:
* It is allowed to use `using namespace std` to import the `std` namespace
* it is allowed to shorten long names using `namespace short_name = long_name`

Reply:
Lol, yes, I thought of the second only later, of course.